operator_tracker/src/main.rs

```diff
@@ -0,0 +1,178 @@
+
+    let pool = PgPoolOptions::new()
+        .max_connections(5)
+        .connect(database_url)
+        .await
+        .expect("Failed to create connection pool");
+
+    pool.execute(include_str!("../sql/schema.sql"))
+        .await
+        .expect("Failed to initialize DB");
+
+    let eth_rpc = Arc::new(Provider::<Http>::connect(rpc_url).await);
+
+    let registry_coordinator = RegistryCoordinator::new(registry_coordinator_addr, eth_rpc.clone());
+
+    let state = AppState {
+        pool,
+        registry_coordinator,
+    };
+
+    let listener = tokio::net::TcpListener::bind("0.0.0.0:3000")
+        .await
+        .expect("Failed to bind listener");
+
+    let router = Router::new()
+        .route("/versions", post(post_operator_version))
+        .route("/versions", get(list_operator_versions))
+        .route("/versions/:address", get(get_operator_version))
+        .with_state(state);
+
+    axum::serve(listener, router)
+        .await
+        .expect("Failed to serve app");
+}
+
+async fn post_operator_version(
+    state: State<AppState>,
+    Json(payload): Json<OperatorVersionPayload>,
+) -> axum::http::Response<Body> {
+    debug!(
+        "Received operator version request. Version {}",
+        payload.version
+    );
+
+    match create_or_update_operator_version(&state.pool, &state.registry_coordinator, payload).await
+    {
+        Ok(Some(body)) => {
+            info!("Operator version created successfully");
+
+            let (mut status, serialized_body) = serialize_or_err(&body);
+            if status == StatusCode::OK {
+                status = StatusCode::CREATED;
+            }
+
+            axum::http::Response::builder()
+                .header("Location", format!("/versions/{}", body.address))
+                .status(status)
+                .body(serialized_body)
+                .unwrap_or_default() // Should never fail but dont panic
+        }
+        Ok(None) => {
+            info!("Operator version updated or already matched");
+
+            axum::http::Response::builder()
+                .status(axum::http::StatusCode::NO_CONTENT)
+                .body(Body::empty())
+                .unwrap_or_default() // Should never fail but dont panic
+        }
+        Err(err) => {
+            info!("Operator version already exists");
+            err.into_response()
+        }
+    }
+}
+
+async fn list_operator_versions(state: State<AppState>) -> axum::http::Response<Body> {
+    let rows = match operator_tracker::list_operator_versions(&state.pool).await {
+        Ok(rows) => rows,
+        Err(err) => return err.into_response(),
+    };
+
+    let (status, body) = if rows.is_empty() {
+        (axum::http::StatusCode::NO_CONTENT, Body::empty())
+    } else {
+        serialize_or_err(rows)
+    };
+
+    axum::http::Response::builder()
+        .status(status)
+        .body(body)
+        .unwrap_or_default() // Should never fail but dont panic
+}
+
+async fn get_operator_version(
```

Comment:
There should be a special response if there are no operator versions found, to avoid the following:

<img width="648" alt="image" src="https://github.com/user-attachments/assets/8e1fb7b2-739e-43ee-976d-96235292b692">

Reply:
will return 204 -no response- code